### DISCUSSION TITLE
tsParticles 3.0.0

### DISCUSSION THREAD

**Discussion by matteobruni** [matteobruni/tsparticles#5262] [General]
# tsParticles 3.0.0 Changelog

## BREAKING CHANGES

- Removed all tsParticles `load` methods to keep only a single one using a single `object` parameter
- Removed support for shape deprecated properties like `shape.image`, `shape.polygon`, `shape.stroke`
    - the stroke property can be found in the `particles section`, not in the `shape` object
    - the image and polygon properties, like any other shape, are part of the `shape.options` object
- Stroke now is a single object only like all the other particles properties
- Removed multiline text shape, the text shape now supports that
- Simplified some functions, using objects or removing unused overloads, this is breaking for v2 shapes
- Moved some properties to private since they were public by mistake (this shouldn't affect many, the properties were mainly used in the engine)
- Removed obsolete random properties in options objects
- Removed emitters shapes from the plugin files, every emitters shape now has a dedicated package

## Major Changes

- All the components for Front End frameworks were removed from this repository, each one now has its own repository. This is mainly for maintainability. It's easier to support multiple Front End frameworks versions in a dedicated repository instead of a multipurpose monorepository. Vue.js 2.x and 3.x were kept split because they have many differences.
    - Angular (`@tsparticles/angular`): https://github.com/tsparticles/angular
    - Astro (`@tsparticles/astro`): https://github.com/tsparticles/astro
    - Ember (`@tsparticles/ember`): https://github.com/tsparticles/ember
    - Inferno (`@tsparticles/inferno`): https://github.com/tsparticles/inferno
    - jQuery (`@tsparticles/jquery`): https://github.com/tsparticles/jquery
    - Lit (`@tsparticles/lit`): https://github.com/tsparticles/lit *(WIP)*
    - Preact (`@tsparticles/preact`): https://github.com/tsparticles/preact
    - React (`@tsparticles/react`): https://github.com/tsparticles/react
    - Riot (`@tsparticles/riot`): https://github.com/tsparticles/riot
    - Solid (`@tsparticles/solid`): https://github.com/tsparticles/solid
    - Svelte (`@tsparticles/svelte`): https://github.com/tsparticles/
    - Vue.js 2.x (`@tsparticles/vue2`): https://github.com/tsparticles/vue2
    - Vue.js 3.x (`@tsparticles/vue3`): https://github.com/tsparticles/vue3
    - Web Components (`@tsparticles/webcomponents`): https://github.com/tsparticles/webcomponents
    - WordPress (`@tsparticles/wordpress`): https://github.com/tsparticles/wordpress
- Removed all presets from this repository for a single one (<https://github.com/tsparticles/presets>) in the @tsparticles organization, this will make easier to contribute to specific presets or create new ones. (Renaming them to `@tsparticles/preset-<name>` #3977)
- All the previous packages will be deprecated when v3 will come out. Using only the new naming system is the main focus of v3. All the official packages will have `@tsparticles/` organization in the package name, replacing `tsparticles-` prefix, except for `tsparticles` package which will remain the same.
- Moved editor to its own repository: https://github.com/tsparticles/editor
- Pjs package now supports all the legacy pjs options only. This will make this package a wrapper for old pjs users. You can't use tsParticles options in pjs calls.
- Pjs package is no longer part of `@tsparticles/slim` bundle, only included in `@tsparticles/all`.
- Text shape is no longer part of `@tsparticles/slim` bundle, included in `tsparticles`.
- Emoji shape is now part of `@tsparticles/slim` bundle.

## New Features

- Added event on config added to the engine
- Added flat options to tsparticles-confetti options
- Creating support for effects, like bubble that wasn't a real shape
- Added linear easing
- Created new emitters shapes: Canvas, Path and Polygon
- Created trail effect plugin, this one is a real trail drawn in the canvas, it works also in transparent backgrounds. It requires more resources since it's drawn calculating last N positions and not redrawing a semi-transparent canvas on itself, supports also fade.
- Added pop click interaction
- Added limit mode (breaking changes on limit options)
- Added possibility to replace color and opacity for emitters shapes
- Added more options for customizing noises values (only Simplex and Perlin paths)
- Added curl noise path plugin
- Created Simplex and Perlin noise packages from their path plugins, they can be used in multiple packages without duplication (Curl Noise path for example)
- Added new emoji shape, better performance than text shape
- Added clear flag to the root object, enabled by default, if disabled, the canvas won't be cleared

## Bug Fixes

- Improved resize event and density formula

<hr /><em>This discussion was created from the release <a href='https://github.com/tsparticles/tsparticles/releases/tag/v3.0.0'>tsParticles 3.0.0</a>.</em>

**Comment by DezLife** [comment]
This is clearly impossible to use until there is a normal online editor and settings.

**Reply by matteobruni** [reply]
This refers to the website, please open an issue there. https://github.com/tsparticles/website
